pythonforandroid/recipes/android/src/android/_android.pyx

```diff
@@ -280,17 +280,20 @@ class AndroidBrowser(object):
 import webbrowser
 webbrowser.register('android', AndroidBrowser)
 
-cdef extern void android_start_service(char *, char *, char *)
-def start_service(title=None, description=None, arg=None):
-    cdef char *j_title = NULL
-    cdef char *j_description = NULL
-    if title is not None:
-        j_title = <bytes>title
-    if description is not None:
-        j_description = <bytes>description
-    if arg is not None:
-        j_arg = <bytes>arg
-    android_start_service(j_title, j_description, j_arg)
+
+def start_service(title=None, description=None, arg=None,
+                  as_foreground=True):
+    from jnius import autoclass
+    mActivity = autoclass('org.kivy.android.PythonActivity').mActivity
```

Comment:
Also already available at module level.
https://github.com/kivy/python-for-android/blob/v2019.07.08/pythonforandroid/recipes/android/src/android/_android.pyx#L164
That probably requires some refactor though :smile:

Reply:
yeah I've kept the `mActivity` for now, seems a bit ugly to have that outside the function and rely on that being available